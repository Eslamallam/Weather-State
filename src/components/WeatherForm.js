import React, { useState, useEffect } from "react";
import { CountryDropdown, RegionDropdown } from "react-country-region-selector";
import ax from "axios";
import WeatherDetails from "./WeatherDetails";
import "../styles/main.css";

const WeatherForm = () => {
  const [country, setCountry] = useState("");
  const [region, setRegion] = useState("");
  const [weatherDetails, setWeatherDetails] = useState(null);
  const [error, setError] = useState(false);

  const handleCountryChange = val => {
    setCountry(val);
  };

  const handleRegionChange = val => {
    setRegion(val);
  };

  useEffect(() => {
    if (region) {
      ax.get(
        `https://api.openweathermap.org/data/2.5/weather?q=${region}&units=metric&appid=cdaec7ddf2cad8f52e2af566ea8ed0a0`
      )
        .then(res => {
          setError(false);
          setWeatherDetails(res.data);
          console.log(res.data);
        })
        .catch(err => {
          if (
            (err && err.response.status === 404) ||
            (err && err.response.status === 500)
          ) {
            setError(true);
          }
        });
    }
  }, [region]);

  return (
    <div className="weather-box">
      <h5>Select below to get weather state</h5>
      <div className="select-country">
        <CountryDropdown
          value={country}
          onChange={val => handleCountryChange(val)}
        />
      </div>
      <div className="select-region">
        <RegionDropdown
          country={country}
          value={region}
          onChange={val => handleRegionChange(val)}
        />
      </div>
      <div className="details">
        {error ? (
          <div className="error">
            <p>Sorry, something went wrong</p>
          </div>
        ) : (
          weatherDetails && <WeatherDetails details={weatherDetails} />
        )}
      </div>
    </div>
  );
};

export default WeatherForm;
