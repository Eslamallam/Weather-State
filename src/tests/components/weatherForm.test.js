import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import toJSON from "enzyme-to-json";
import WeatherForm from "../../components/WeatherForm";
import details from "../fixtures/weatherDetails";

configure({ adapter: new Adapter() });

test("should render weather form correctly", () => {
  const wrapper = shallow(<WeatherForm />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should render weather form correctly with data", () => {
  const wrapper = shallow(<WeatherForm details={details} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});
