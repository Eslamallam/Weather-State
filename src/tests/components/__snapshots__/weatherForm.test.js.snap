// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render weather form correctly 1`] = `
<div
  className="weather-box"
>
  <h5>
    Select below to get weather state
  </h5>
  <div
    className="select-country"
  >
    <CountryDropdown
      blacklist={Array []}
      classes=""
      defaultOptionLabel="Select Country"
      disabled={false}
      id=""
      labelType="full"
      name="rcrs-country"
      onBlur={[Function]}
      onChange={[Function]}
      priorityOptions={Array []}
      showDefaultOption={true}
      value=""
      valueType="full"
      whitelist={Array []}
    />
  </div>
  <div
    className="select-region"
  >
    <RegionDropdown
      blankOptionLabel="-"
      classes=""
      country=""
      countryValueType="full"
      customOptions={Array []}
      defaultOptionLabel="Select Region"
      disableWhenEmpty={false}
      disabled={false}
      id=""
      labelType="full"
      name="rcrs-region"
      onBlur={[Function]}
      onChange={[Function]}
      showDefaultOption={true}
      value=""
      valueType="full"
    />
  </div>
  <div
    className="details"
  />
</div>
`;

exports[`should render weather form correctly with data 1`] = `
<div
  className="weather-box"
>
  <h5>
    Select below to get weather state
  </h5>
  <div
    className="select-country"
  >
    <CountryDropdown
      blacklist={Array []}
      classes=""
      defaultOptionLabel="Select Country"
      disabled={false}
      id=""
      labelType="full"
      name="rcrs-country"
      onBlur={[Function]}
      onChange={[Function]}
      priorityOptions={Array []}
      showDefaultOption={true}
      value=""
      valueType="full"
      whitelist={Array []}
    />
  </div>
  <div
    className="select-region"
  >
    <RegionDropdown
      blankOptionLabel="-"
      classes=""
      country=""
      countryValueType="full"
      customOptions={Array []}
      defaultOptionLabel="Select Region"
      disableWhenEmpty={false}
      disabled={false}
      id=""
      labelType="full"
      name="rcrs-region"
      onBlur={[Function]}
      onChange={[Function]}
      showDefaultOption={true}
      value=""
      valueType="full"
    />
  </div>
  <div
    className="details"
  />
</div>
`;
